android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java

```diff
@@ -257,6 +252,11 @@ private void clampForMinimumVisibility(WindowManager.LayoutParams params) {
 
     @Override
     public void onCreate() {
+        FlutterEngine flutterEngine = FlutterEngineCache.getInstance().get(OverlayConstants.CACHED_TAG);
+        final DartExecutor dartExecutor =  flutterEngine.getDartExecutor();
+        flutterChannel = new MethodChannel(dartExecutor, OverlayConstants.OVERLAY_TAG);
+        overlayMessageChannel = new BasicMessageChannel(dartExecutor, OverlayConstants.MESSENGER_TAG, JSONMessageCodec.INSTANCE);
```

Comment:
This setup was previously done on lines 53/54 as final class variables. I've moved it here in preparation for handling the 'null engine' scenario which can happen when the service is restarted. Since I have disabled restarting I haven't included the handling of null engine for now but it still makes sense to have this here rather than in the private class variable.